### PR TITLE
Story/ccp 116

### PR DESCRIPTION
**Pull request by mkarthik90**
[CCP-116] Changes done for loading the synonym to tables. Included method for JSON building and iterating in JSP.

### CODE REVIEW

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptWrapperAddController.java [hunk 1]
```diff
@@ -174,12 +177,43 @@ public String search(HttpServletRequest req, ModelMap model) {
...
+        jsonStringBuilder.append(":");
+        jsonStringBuilder.append(synonyms.size());
+        jsonStringBuilder.append(",");
+        jsonStringBuilder.append("\"synonyms\":");
+        jsonStringBuilder.append("[");
+        for (ConceptEntry syn : synonyms) {
+            // Appending for next element in JSON
+            if (i != 0) {
+                jsonStringBuilder.append(",");
+            }
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptWrapperAddController.java]: You should be able to to simply always append a comma, and remove this condition.

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptWrapperAddController.java [hunk 2]
```diff
@@ -174,12 +177,43 @@ public String search(HttpServletRequest req, ModelMap model) {
...
+            jsonStringBuilder.append("{");
+            jsonStringBuilder.append("\"id\":\"" + syn.getId() + "\"");
+            jsonStringBuilder.append(",");
+            jsonStringBuilder.append("\"word\":\"" + syn.getWord() + "\"");
+            jsonStringBuilder.append(",");
+            jsonStringBuilder.append("\"description\":\"" + syn.getDescription().replaceAll("\"", "'") + "\"");
+            jsonStringBuilder.append(",");
+            String pos = syn.getPos().replaceAll("\"", "'");
+            jsonStringBuilder.append("\"pos\":\"" + pos + "\"");
+            jsonStringBuilder.append("}");
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptWrapperAddController.java]: just always have a comma here.

**Comment by jdamerow** [comment]
ok to test

**Comment by jdamerow** [comment]
hmm, that doesn't look right:
image

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptWrapperAddController.java (hunk 1 above)
Comment (mkarthik90) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptWrapperAddController.java]: If i remove that condition and append , for all the values i am getting JSON parsing error.

**Comment by mkarthik90** [comment]
@jdamerow Please test this change in Add New Wordnet concept wrapper. I tested from this screen and it worked fine. This issue you have specified is fixed in a different story, I will provide the story number for this issue.

**Comment by mkarthik90** [comment]
@jdamerow  The table with undefined attributes is because of another issue and it has been fixed in story [CCP-117]

**Comment by jdamerow** [comment]
ok to test

**Comment by jdamerow** [comment]
For some reason, I can add synonyms like pony  or horse, but I can't add a synonym, when I search for "test". The "add" link doesn't do anything.
1. create concept wrapper
2. click "add synonym"
3. search for 'test'
4. => can't "add"

**Pull request closed without merging**